manifests/v1beta1/components/cert-generator/cert-generator.yaml

```diff
@@ -12,10 +12,14 @@ spec:
         sidecar.istio.io/inject: "false"
     spec:
       serviceAccountName: katib-cert-generator
+      securityContext:
+        runAsNonRoot: true
```

Comment:
Also, why do we need this ?

Reply:
Generally, containers should run as a non-root user. Fortunately,  katib-cert-generator does not have any features that must use privileges. So, I set true in securityContext.runAsNonRoot filed.

By the way, other components(ex: katib-controller) use the alpine image and do not prohibit running as a root user. I consider we should use the distroless image and do not permit running as a root user.

WDYT @andreyvelich ?

Reply:
In `distroless` image can we execute in the container to debug it ?
Do you know what is security issues are we getting by running these containers as root ?
I thought we should be consistent across Katib components.

cc @gaocegege @johnugeorge

Reply:
> In distroless image can we execute in the container to debug it ?

Sorry, I thought we could debug the container using `kubectl debug` command, but I found an ephemeral debug container is an alpha feature.

https://kubernetes.io/docs/tasks/debug-application-cluster/debug-running-pod/#ephemeral-container

So, I think we could reconsider distroless images when is ephemeral debug containers feature is available by default.

> Do you know what is security issues are we getting by running these containers as root ?

I can not show you anything other than the security recommendation in  K8s documentation.

https://kubernetes.io/docs/tasks/administer-cluster/securing-a-cluster/#controlling-what-privileges-containers-run-with

Reply:
> So, I think we could reconsider distroless images when is ephemeral debug containers feature is available by default.

Maybe, let's create an issue to track it ? We might have to add `runAsNonRoot: true`  for all of our Katib components in the future.

Reply:
It sounds good!
In this PR, I'd like to remove securityContext field. In addition, I'd like to create an issue about it.

Reply:
I created an Issue.
https://github.com/kubeflow/katib/issues/1664

Reply:
Thank you @tenzen-y!

Reply:
I removed the securityContext field!